tests/phpunit/tests/theme/wpThemeJsonResolver.php

```diff
@@ -629,23 +644,41 @@ function test_get_user_data_from_wp_global_styles_does_not_use_uncached_queries(
 			WP_Theme_JSON_Resolver::clean_cached_data();
 		}
 		$query_count = count( $this->queries ) - $query_count;
-		$this->assertSame( 0, $query_count, 'Unexpected SQL queries detected for the wp_global_style post type' );
+		$this->assertSame( 0, $query_count, 'Unexpected SQL queries detected for the wp_global_style post type prior to creation.' );
 
 		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme );
-		$this->assertEmpty( $user_cpt );
+		$this->assertEmpty( $user_cpt, 'User CPT is expected to be empty.' );
 
 		$user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme, true );
-		$this->assertNotEmpty( $user_cpt );
+		$this->assertNotEmpty( $user_cpt, 'User CPT is expected not to be empty.' );
 
 		$query_count = count( $this->queries );
 		for ( $i = 0; $i < 3; $i ++ ) {
 			$new_user_cpt = WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles( $theme );
 			WP_Theme_JSON_Resolver::clean_cached_data();
-			$this->assertSameSets( $user_cpt, $new_user_cpt );
+			$this->assertSameSets( $user_cpt, $new_user_cpt, "User CPTs do not match on run {$i}." );
+		}
+		$query_count = count( $this->queries ) - $query_count;
+		$this->assertSame( 1, $query_count, 'Unexpected SQL queries detected for the wp_global_style post type after creation.' );
```

Comment:
This has increased from zero to one.

As the transient has been removed, it's no longer set on post creation so a single query is made on the following call (in run 0), in subsequent runs, it is cached.